web-registry/src/hooks/useBasketDetails.ts

```diff
@@ -0,0 +1,110 @@
+import { useState, useEffect } from 'react';
+
+import { getBasket, getBasketBalances, getBatchInfo } from '../lib/ecocredit';
+import { IBasket, BasketBalance, IBatchInfo } from '../types/ledger/ecocredit';
+import { BasketOverviewProps } from '../components/organisms';
+
+interface BasketExtended {
+  classes: string[];
+  description: string;
+  curator: string;
+  totalAmount: number;
+  // displayDenom ?
+}
+
+interface BasketDetailsAll {
+  basket: IBasket | undefined;
+  basketExtended: BasketExtended;
+  basketBalances: BasketBalance[];
+  batchesInfo: IBatchInfo[];
+}
+
+type BasketDetailsData = {
+  dataOverview: BasketOverviewProps | null;
+  dataBasketBatches: IBatchInfo[] | undefined;
+};
+
+const useBasketDetails = (
+  basketDenom: string | undefined,
+): BasketDetailsData => {
+  const [data, setData] = useState<BasketDetailsAll | null>(null);
+  const [dataOverview, setDataOverview] = useState<BasketOverviewProps | null>(
+    null,
+  );
+
+  // fetch all the data
+  useEffect(() => {
+    if (basketDenom === undefined) return;
+
+    async function fetchData(basketDenom: string): Promise<void> {
+      try {
+        const basket = await getBasket(basketDenom);
+        // // TODO: Display basket denom
+        // const displayDenom = 'eco.C.rNCT';
+        // TODO: Basket description (see comment in Figma)
+        // https://www.figma.com/file/x5vjWsddiUBzP2N13AFOPw?node-id=32:10028#155844414
+        // const description = getBasketDescription(basketDenom);
+        const description =
+          'The Regen Nature Carbon Ton groups together carbon sequestration ecocredits into one tradeable asset. Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
+        // TODO: Basket Curator
+        const curator = 'Regen Network Development, Inc';
+        const balances = await getBasketBalances(basketDenom);
+        const batches: IBatchInfo[] = [];
+        balances.balances.map(async batchBalance => {
+          getBatchInfo(batchBalance.batchDenom).then(batchInfo => {
+            batches.push(batchInfo.info);
+          });
+        });
+
+        setData({
+          basket: basket.basket,
+          basketExtended: {
+            classes: basket.classes,
+            description,
+            curator,
+            totalAmount: 0,
+          },
+          basketBalances: balances.balances,
+          batchesInfo: batches,
+        });
+      } catch (error) {
+        // TODO Manage error
+      }
+    }
+
+    fetchData(basketDenom);
+  }, [basketDenom]);
+
+  // Prepare data for BasketOverview
+  useEffect(() => {
+    if (data?.basket) {
+      const _dataOverview: BasketOverviewProps = {
```

Comment:
I think it comes from when I used Python... I use it when the variable represents the "preparation" of a data, something in the process, yes within a delimited scope, or something like that. In this case the final data is called dataOverview (managed by the useState hook). I would have used the same name in the effect hook... so just prefix it with an underscore! I am also taking note of these little details so that over time we can precisely define our own style guide, etc.